src/server/scripts/Northrend/Naxxramas/boss_thaddius.cpp

```diff
@@ -560,23 +1101,97 @@ class spell_thaddius_polarity_shift : public SpellScriptLoader
         }
 };
 
-class achievement_polarity_switch : public AchievementCriteriaScript
+class spell_thaddius_magnetic_pull : public SpellScriptLoader
 {
     public:
-        achievement_polarity_switch() : AchievementCriteriaScript("achievement_polarity_switch") { }
+        spell_thaddius_magnetic_pull() : SpellScriptLoader("spell_thaddius_magnetic_pull") { };
+
+        class spell_thaddius_magnetic_pull_SpellScript : public SpellScript
+        {
+            PrepareSpellScript(spell_thaddius_magnetic_pull_SpellScript);
+
+            bool Validate(SpellInfo const*) override
+            {
+                return sSpellMgr->GetSpellInfo(SPELL_MAGNETIC_PULL) ? true : false;
+            }
+
+            void HandleCast() // only feugen ever casts this according to wowhead data
+            {
+                Unit* feugen = GetCaster();
+                if (!feugen || feugen->GetEntry() != NPC_FEUGEN)
+                    return;
+                
+                Unit* stalagg = ObjectAccessor::GetCreature(*feugen, feugen->GetInstanceScript()->GetGuidData(DATA_STALAGG));
+                if (!stalagg)
+                    return;
+
+                Unit* feugenTank = feugen->GetVictim();
+                Unit* stalaggTank = stalagg->GetVictim();
+
+                if (!feugenTank || !stalaggTank || feugenTank == stalaggTank)
+                    return;
+
+                ThreatManager& feugenThreat = feugen->getThreatManager();
+                ThreatManager& stalaggThreat = stalagg->getThreatManager();
+
+                float feugenTankThreat = feugenThreat.getThreat(feugenTank);
+                float feugenOtherThreat = feugenThreat.getThreat(stalaggTank);
+                float stalaggTankThreat = stalaggThreat.getThreat(stalaggTank);
+                float stalaggOtherThreat = stalaggThreat.getThreat(feugenTank);
+
+                // set the two entries in feugen's threat table to be equal to the ones in stalagg's
+                feugenThreat.addThreat(stalaggTank, stalaggTankThreat - feugenOtherThreat);
+                feugenThreat.addThreat(feugenTank, stalaggOtherThreat - feugenTankThreat);
+
+                // set the two entries in stalagg's threat table to be equal to the ones in feugen's
+                stalaggThreat.addThreat(feugenTank, feugenTankThreat - stalaggOtherThreat);
+                stalaggThreat.addThreat(stalaggTank, feugenOtherThreat - stalaggTankThreat);
+
+                // pull the two tanks across
+                feugenTank->CastSpell(stalaggTank, SPELL_MAGNETIC_PULL_EFFECT);
+                stalaggTank->CastSpell(feugenTank, SPELL_MAGNETIC_PULL_EFFECT);
+
+                // and make both attack their respective new tanks
+                if (feugen->GetAI())
+                    feugen->GetAI()->AttackStart(stalaggTank);
```

Comment:
What'd I use if I want to change victim instead (regardless of someone else being higher than them in threat).